torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -249,28 +248,36 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                     node.replace_all_uses_with(replacement_proxy.node)
                     graph.erase_node(node)
 
-    # DCE symbols (which are guaranteed to be pure) only
-    graph.eliminate_dead_code()
-
-    # Now do one more pass that specializes all symfloats we didn't manage
-    # to tensorify away.
     nodes = list(graph.nodes)
-    for i, node in enumerate(nodes[:-1]):
+    for i, node in enumerate(nodes[-2::-1]):
         with graph.inserting_before(
-            nodes[i + 1] if node not in placeholders else first_non_placeholder
+            nodes[-(i + 2)] if node not in placeholders else first_non_placeholder
         ):
+            if (
+                len(node.users) == 0
+                and node.op != "placeholder"
+                and node.op != "output"
+                and not node.is_impure()
+            ):
+                graph.erase_node(node)
+                continue
+
             args = []
             kwargs: Dict[str, Any] = {}
             transform = False
             for a in node.args:
-                if a in specialized_float_nodes:
-                    args.append(specialized_float_nodes[a])
-                elif (
+                if (
                     isinstance(a, fx.Node)
                     and "val" in a.meta
                     and isinstance(zf := a.meta["val"], torch.SymFloat)
-                    and isinstance(zf.node.expr, Symbol)
                     and zf.node.hint is not None
+                    and all(
+                        (p := expr_to_sym_proxy.get(s)) is not None
+                        and p.node is not None
+                        and "val" in p.node.meta
+                        and isinstance(p.node.meta["val"], torch.SymFloat)
```

Comment:
This seems like a circuitous way of working out the symbol is a float symbol. Instead, just use torch.utils._sympy.symbol.symbol_is_type